adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryRawFileRepository.java

```diff
@@ -69,15 +69,15 @@ public RawFileInfo create(RawFileInfo fileInfo) {
         CsvMapper mapper = new CsvMapper();
         CsvSchema schema = CsvSchema.emptySchema().withHeader();
 
-        Collection<String> toReturn = new HashSet<>();
+        Collection<String> toReturn = new ArrayList<>();
 
         try {
             ((CsvSchema) (mapper.readerFor(Map.class)
                     .with(schema)
                     .readValues(csvFile).getParser().getSchema())).iterator().forEachRemaining(column -> toReturn.add(column.getName()));
         } catch (IOException e) {
             //TODO: this should go back up to use case level so it can be properly reported
-            return Collections.emptySet();
+            return Collections.emptyList();
```

Comment:
I'd suggest adding a note to the Javadoc that says that the returned list will contain all headers present in the CSV file, including any duplicates.

Reply:
Actually, thinking more, given that this actually affects program correctness, it's probably better to change the return value explicitly to a `List` (in addition to documenting in the Javadoc). For example, you wouldn't want someone to change the implementation of the method internally back to a `Set` in the future, thinking that it still satisfies the `Collection` interface, because E43 would never find any duplicate columns. Changing the return type, IMHO, is a more explicit statement that there is a reason why a `List` is used here.